keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -514,310 +481,126 @@ spec:
-field in each `ResourceSlice` object is used to determine which objects are
-managed by which kubelet.
-
-**NOTE:**  `ResourceSlices` are published separately for each driver, using
-whatever version of the `resource.k8s.io` API is supported by the kubelet. That
-same version is then also used in the gRPC interface between the kubelet and
-the DRA drivers providing content for those objects. It might be possible to
-support version skew (= keeping kubelet at an older version than the control
-plane and the DRA drivers) in the future, but currently this is out of scope.
-
-Embedded inside each `ResourceSlice` is the representation of the resources
-managed by a driver according to a specific "structured model". In the example
-seen below, the structured model in use is called `namedResources`:
-
-```yaml
-kind: ResourceSlice
-apiVersion: resource.k8s.io/v1alpha2
-...
-spec:
-  nodeName: worker-1
-  driverName: cards.dra.example.com
-  namedResources:
-    ...
-```
-
-Such a model could be created to represent resources in a manner similar to the
-opaque strings passed over the tradition device plugin API to the kubelet. The
-one addition being that each named resource can have a set of arbitrary
-attributes attached to it.
-
-If a driver wanted to use a different structured model to represent its resources,
-a new structured model would need to be defined inside Kuberenetes, and a field
-would need to be added to this struct at the same level as
-`namedResources`. Driver implementors would then have the option
-to set this new field instead.
-
-**Note:** If a new model is added to the schema but clients are not updated,
-they'll encounter an object with no information from any known structured model
-when they serialize into their known version of a `ResourceSlice`. This
-tells them that they cannot handle the object because the API has been extended.
-
-Drivers can use different structured models by publishing multiple
-`ResourceSlice` objects, as long as each model represents a distinct set of
-resources. Whether the information about resources of one particular structured
-model must fit into one ResourceSlice object (or be distributed across
-many) depends on how that particular structured model describes its resources. In
-all cases, the size of each object is a hard limit and one must take this into
-account when designing a structured model and preparing ResourceSlice objects
-for it.
-
-Below is an example of a driver that provides two discrete GPU cards using the
-`namedResources` model described above:
+In the kubelet case, each driver running on a node publishes a set of
+`ResourceSlice` objects to the API server for its own resources, using its
+connection to the apiserver. The collection of these objects form a pool from
+which resources can be allocated. Some additional fields (defined in the API
+section) enable a consumer to determine whether it has a complete and
+consistent view of that pool.
```

Comment:
Given the description here, wouldn't `ResourcePoolSlice` be a more appropriate name? This is one slice of an overall `ResourcePool`.

Reply:
That is a renaming that makes sense to me.

Reply:
@thockin @johnbelamaric what do you think?

Reply:
Sounds good. I don't feel too strongly. I just keep getting hung up on seeing this as described here:
https://github.com/kubernetes/enhancements/pull/4384#discussion_r1482130696